backend/src/main/java/gov/cdc/usds/simplereport/api/patient/PatientDataResolver.java

```diff
@@ -7,51 +7,80 @@
 import gov.cdc.usds.simplereport.db.model.Person;
 import gov.cdc.usds.simplereport.db.model.PhoneNumber;
 import gov.cdc.usds.simplereport.db.model.TestEvent;
-import gov.cdc.usds.simplereport.service.dataloader.PatientLastTestDataLoader;
-import gov.cdc.usds.simplereport.service.dataloader.PatientPhoneNumbersDataLoader;
-import gov.cdc.usds.simplereport.service.dataloader.PatientPrimaryPhoneDataLoader;
-import graphql.kickstart.tools.GraphQLResolver;
-import graphql.schema.DataFetchingEnvironment;
+import gov.cdc.usds.simplereport.db.model.auxiliary.PersonName;
+import gov.cdc.usds.simplereport.db.repository.PhoneNumberRepository;
+import gov.cdc.usds.simplereport.db.repository.TestEventRepository;
 import java.util.List;
+import java.util.Map;
+import java.util.UUID;
 import java.util.concurrent.CompletableFuture;
-import org.springframework.stereotype.Component;
-
-@Component
-public class PatientDataResolver
-    implements GraphQLResolver<Person>, PersonNameResolver<Person>, InternalIdResolver<Person> {
-
-  private final PatientPrimaryPhoneDataLoader _patientPrimaryPhoneDataLoader;
-  private final PatientPhoneNumbersDataLoader _patientPhoneNumbersDataLoader;
-  private final PatientLastTestDataLoader _patientLastTestDataLoader;
-
-  PatientDataResolver(
-      PatientPrimaryPhoneDataLoader patientPrimaryPhoneDataLoader,
-      PatientPhoneNumbersDataLoader patientPhoneNumbersDataLoader,
-      PatientLastTestDataLoader patientLastTestDataLoader) {
-    _patientPrimaryPhoneDataLoader = patientPrimaryPhoneDataLoader;
-    _patientPhoneNumbersDataLoader = patientPhoneNumbersDataLoader;
-    _patientLastTestDataLoader = patientLastTestDataLoader;
-  }
+import java.util.stream.Collectors;
+import org.dataloader.DataLoader;
+import org.springframework.graphql.data.method.annotation.SchemaMapping;
+import org.springframework.graphql.execution.BatchLoaderRegistry;
+import org.springframework.stereotype.Controller;
+import reactor.core.publisher.Mono;
+
+@Controller
+public class PatientDataResolver implements PersonNameResolver<Person>, InternalIdResolver<Person> {
+
+  public PatientDataResolver(
+      BatchLoaderRegistry registry,
+      TestEventRepository testEventRepository,
+      PhoneNumberRepository phoneNumberRepository) {
+    registry
+        .forTypePair(UUID.class, TestEvent.class)
+        .withName("patientLastTestLoader")
+        .registerMappedBatchLoader(
+            (patientIds, batchLoaderEnvironment) -> {
+              Map<UUID, TestEvent> found =
+                  testEventRepository.findLastTestsByPatient(patientIds).stream()
+                      .collect(Collectors.toMap(TestEvent::getPatientInternalID, s -> s));
+              return Mono.just(found);
+            });
+
+    Class<List<PhoneNumber>> phoneNumberListClazz = (Class) List.class;
```

Comment:
this is some weird java quark, since you cant simply do `List.class<PhoneNumber.class>`

Reply:
wow that's fun, is the use of `zz` just generally accepted for this kind of definition?

Reply:
https://stackoverflow.com/questions/2529974/why-do-java-programmers-like-to-name-a-variable-clazz